repository/src/main/java/org/alfresco/repo/search/impl/AbstractCategoryServiceImpl.java

```diff
@@ -546,6 +550,57 @@ public void deleteClassification(StoreRef storeRef, QName aspectName)
 
     public abstract List<Pair<NodeRef, Integer>> getTopCategories(StoreRef storeRef, QName aspectName, int count);
 
+    /**
+     * Creates search query parameters used to get top categories.
+     * Can be used as a base both wih SOLR and ES.
+     * @param storeRef Node store reference
+     * @param aspectName Aspect name. "cm:generalclassifiable" aspect should be used for usual cases.
+     *                   It is possible to use a custom aspect but it must have valid category property
+     * @param count Will be used as faceted results limit, when system has very many categories this must be reflecting that number
+     * @return SearchParameters to perform search for top categories.
+     */
+    protected SearchParameters createSearchTopCategoriesParameters(StoreRef storeRef, QName aspectName, int count) {
+        final AspectDefinition aspectDefinition = dictionaryService.getAspect(aspectName);
+        if(aspectDefinition == null)
+        {
+            throw new IllegalStateException("Unknown aspect");

```

Comment:
Maybe IllegalArgumentException?

Reply:
I didn't like that one too but I wanted to be backwards compatible and that particular part is taken from `SolrCategoryService` class. TBH, I didn't find any useful usage of `getTopCategories` (I think `jscript` part is not useed anywhere) but it may be hidden somehow I can't find it and I am just cautious...
Do you think it is safe to change the type of Exception?